### ISSUE TITLE
Connecting with standard MySQL?

### ISSUE THREAD

**Issue by pumplerod** [groue/GRDB.swift#195]
I'm developing an app using GRDB and sqlite for the client and am now at the stage where I need to build a server where some of the client data will get transferred to and from.  The server is built with Perfect 2.0 and I was planning on constructing that with MySQL.

Will GRDB.swift work with a standard MySQL server?  I'm concerned SQLite will not be robust enough for the server side, however I would love to keep my code consistent between client and server, so it would be ideal to use GRDB.swift on the server as well.

I'm not trying to create a direct connection to the server db.  I'm building a REST API to deal with the requests and process the data on the server side.

**Comment by groue** [comment]
Hello @pumplerod 

> The server is built with Perfect 2.0 and I was planning on constructing that with MySQL.

Sounds both interesting and sensible.

> I'm not trying to create a direct connection [from the client] to the server db. I'm building a REST API to deal with the requests and process the data on the server side [which has direct connection to the db].

Understood.

> Will GRDB.swift work with a standard MySQL server? [...] I would love to keep my code consistent between client and server, so it would be ideal to use GRDB.swift on the server as well.

Thanks for your interest!

GRDB provides applications with tools that only SQLite supports: [notification of database changes](https://github.com/groue/GRDB.swift#database-changes-observation), [notification of changes in the results of request](https://github.com/groue/GRDB.swift#fetchedrecordscontroller), [customizable full-text search](https://github.com/groue/GRDB.swift/blob/master/Documentation/FTS5Tokenizers.md), [custom SQL functions](https://github.com/groue/GRDB.swift#custom-sql-functions), [unicode support that doesn't suck](https://github.com/groue/GRDB.swift#unicode), etc. Finally, the GRDB [concurrency models](https://github.com/groue/GRDB.swift#concurrency) have been designed 1: for multi-threaded applications, 2: around SQLite, and 3: purposely ignore the constraints of concurrent connections that any server has to support.

**I intend to keep my free time dedicated at preserving the focus of this library on applications**, which means SQLite. SQLite is super cool for applications.

Now, everybody can fork GRDB, perform a delicate surgery, and have it support servers, and MySQL. [Pierlis](http://pierlis.com), the company I work for, could contract this service for you and your company. But that initial effort would not be enough to maintain a high level of quality over time. **GRDB on the server would need one or several sponsors.** We can start a discussion about this topic, if you want.

Did you make any research about the landscape of serious Swift layers over MySQL?

- Perfect has a [MySQL connector](http://perfect.org/docs/MySQL.html) that has a very important quality: it has been validated by the Perfect team.
- IBM is working on a SQL query builder, [Swift-Kuery](https://github.com/IBM-Swift/Swift-Kuery), which has been developed for [Kitura](http://www.kitura.io), but should integrate well with Perfect.
- [Vapor's Fluent](https://vapor.github.io/documentation/fluent/driver.html) may also work with Perfect.
- There surely are other tools.
- And finally there is [Realm](http://realm.io). I guess you'd have immediate code sharing between your client and your server with Realm.

If none of those tools are as nice as GRDB, make sure you tell them about it :-)

**Comment by pumplerod** [comment]
Thanks for the reply.

> Perfect has a MySQL connector that has a very important quality: it has been validated by the Perfect team.

I will most likely go with the PerfectlySwift solution.  I've just come to very much appreciate your library and thought it would be an added bonus to keep the code the same between server/client.  But I'm already attempting the server with Swift instead of PHP so I guess I should count my blessing.

I don't have the coding chops yet to fork and customize GRDB and I'm just a lone code hobbiest so I'm afraid I've got to look for cost effective solutions for the time being.

> If none of those tools are as nice as GRDB, make sure you tell them about it :-)

I've learned quite a bit just by reading through your code and materials so I'm a huge fan.  I'll sing your praises to whomever I can.
